src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -169,52 +164,67 @@ internal static TimeSpan ResolveTimeSpanFromMilliseconds(string nameOfProperty,
-                    attribute.ConnectionStringSetting));
+                    $"The CosmosDBTrigger connection string must be set either via a '{Constants.DefaultConnectionStringName}' IConfiguration connection string, via the {attributeProperty} property or via {optionsProperty}.");
             }
 
+            return connectionString;
+        }
+
+        private string ResolveAttributeLeasesConnectionString(CosmosDBTriggerAttribute attribute)
+        {
+            string connectionString = ResolveConnectionString(attribute.LeaseConnectionStringSetting, nameof(CosmosDBTriggerAttribute.LeaseConnectionStringSetting));
+
             if (string.IsNullOrEmpty(connectionString))
             {
+                string attributeProperty = $"{nameof(CosmosDBTriggerAttribute)}.{nameof(CosmosDBTriggerAttribute.LeaseConnectionStringSetting)}";
+                string optionsProperty = $"{nameof(CosmosDBOptions)}.{nameof(CosmosDBOptions.ConnectionString)}";
                 throw new InvalidOperationException(
-                    string.Format(CultureInfo.CurrentCulture,
-                    "The CosmosDBTrigger connection string must be set either via a '{0}' app setting, via the CosmosDBTriggerAttribute.ConnectionStringSetting property or via DocumentDBConfiguration.ConnectionString.",
-                    CosmosDBExtensionConfigProvider.AzureWebJobsCosmosDBConnectionStringName));
+                    $"The CosmosDBTrigger lease connection string must be set either via a '{Constants.DefaultConnectionStringName}' IConfiguration connection string, via the {attributeProperty} property or via {optionsProperty}.");
             }
 
             return connectionString;
         }
 
-        private string ResolveAttributeLeasesConnectionString(CosmosDBTriggerAttribute attribute, string triggerConnectionString)
+        internal string ResolveConnectionString(string unresolvedConnectionString, string propertyName)
         {
-            // If the leases connection is not specified, it connects to the monitored service
-            string connectionString = string.IsNullOrEmpty(this._leasesConnectionString) ? triggerConnectionString : this._leasesConnectionString;
-            if (!TryReadFromSettings(attribute.LeaseConnectionStringSetting, connectionString, out connectionString))
+            // First, resolve the string.
+            if (!string.IsNullOrEmpty(unresolvedConnectionString))
             {
-                throw new InvalidOperationException(
-                    string.Format(CultureInfo.CurrentCulture,
-                    "The CosmosDBTriggerAttribute.LeaseConnectionStringSetting '{0}' property specified does not exist in the Application Settings.",
-                    attribute.ConnectionStringSetting));
+                string resolvedString = _nameResolver.Resolve(unresolvedConnectionString);
```

Comment:
Need to discuss this. Not sure the resolution should go through the name resolver, but at the very least, it should still fallback to a connection string lookup. Ideally, this would be common logic reused across all extensions.

Reply:
We could go directly against `IConfiguration` -- but all other bindings use `INameResolver` and this isn't a change here, it's just been relocated. It *does* fallback to the connection string. That's because it'll return the _options.ConnectionString if this is missing. And that is set up with the ConnectionString.

Reply:
Actually, now I get what you're saying. I'm going to swap this to go directly against IConfiguration and check ConnectionStrings first. I really want to [obsolete] INameResolver so we can eventually remove it. We'll need to port the ResolveWholeString() behavior for resolving %% in strings, but that shouldn't be tough.